civis/base.py

```diff
@@ -134,22 +135,29 @@ class CivisAsyncResultBase(futures.Future):
     polling_interval : int or float
         The number of seconds between API requests to check whether a result
         is ready.
-    api_key : str, optional
+    api_key : DEPRECATED str, optional
         Your Civis API key. If not given, the :envvar:`CIVIS_API_KEY`
         environment variable will be used.
+    client : :class:`civis.APIClient`, optional
+        If not provided, an :class:`civis.APIClient` object will be
+        created from the :envvar:`CIVIS_API_KEY`.
     poll_on_creation : bool, optional
         If ``True`` (the default), it will poll upon calling ``result()`` the
         first time. If ``False``, it will wait the number of seconds specified
         in `polling_interval` from object creation before polling.
     """
     def __init__(self, poller, poller_args,
-                 polling_interval=None, api_key=None,
+                 polling_interval=None, api_key=None, client=None,
                  poll_on_creation=True):
         super().__init__()
         self.poller = poller
         self.poller_args = poller_args
         self.polling_interval = polling_interval
-        self.api_key = api_key
+        if api_key is not None:
+            warnings.warn('The "api_key" parameter is deprecated and will be '
+                          'removed in v2. Please use the `client` parameter '
+                          'instead.', FutureWarning)
```

Comment:
this warning is in a lot of places, might be nice to create a single function and call that instead.